### PR TITLE
add Transaction concept

### PR DESCRIPTION
**Pull request by brunobuddy**
Create Transaction model and add new API entrypoint : PUT
/api/customers/$id?$transaction_amount=$transaction_amount

### CODE REVIEW

File: app/Http/Controllers/CustomerController.php [hunk 1]
```diff
@@ -21,4 +22,30 @@ public function show($id)
...
+     * Met à jour l'amount de l'utilisateur suite à une transaction ou un refund
+     *
+     * @ returns updated Customer amount
+     */
+    public function update($id)
+    {
+        $customer = Customer::findOrFail($id);
+        $transactionAmount = request('transaction_amount');
+
+        // On vérifie que le Customer a assez d'argent en cas de dépense
```
Comment (n1c01a5) [inline app/Http/Controllers/CustomerController.php]: in english

File: app/Http/Controllers/CustomerController.php [hunk 2]
```diff
@@ -21,4 +22,30 @@ public function show($id)
...
+        if ($transactionAmount < 0 && $customer->amount < $transactionAmount * -1) {
+            return abort(403, 'Insufficient funds');
```
Comment (n1c01a5) [inline app/Http/Controllers/CustomerController.php]: simplier is `$customer->amount >= $transactionAmount` no?

File: app/Http/routes.php [hunk 3]
```diff
@@ -15,9 +15,15 @@
     return view('welcome');
 });
 
+// simulate flight delayed
+Route::post('/delay-flight', function () {
+    $customers = \App\Customer::all();
+    \App\Services\ContactService::sendCustomerAppLink($customers, true, true);
+    return "<h1>flight delayed</h1>";
```
Comment (n1c01a5) [inline app/Http/routes.php]: do not returns html code even if for the test

File: app/Services/ContactService.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\Facades\Mail;
+
+class ContactService
+{
+    /*
+     * Envoie le lien de la CustomerApp aux Customers
```
Comment (n1c01a5) [inline app/Services/ContactService.php]: in english

File: app/Services/ContactService.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+     *
+     * @param $customers array
+     * @param byEmail boolean
+     * @param bySms boolean
+     *
+     */
+    public static function sendCustomerAppLink($customers, $byEmail = null, $bySms = null)
```
Comment (n1c01a5) [inline app/Services/ContactService.php]: I think is better to initiate `$byEmail` with false. As this It's a `boolean`.
Idem for `$bySms`.

File: config/settings.php [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/**
+ * Settings de l'app
```
Comment (n1c01a5) [inline config/settings.php]: `App settings`

File: database/migrations/2016_11_26_162123_create_transactions_table.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('transactions', function (Blueprint $table) {
+            $table->increments('id');
+            $table->integer('customer_id')->unsigned();
+            $table->foreign('customer_id')
+                ->references('id')->on('customers')
```
Comment (n1c01a5) [inline database/migrations/2016_11_26_162123_create_transactions_table.php]: new line `->on('customers')`

File: resources/views/welcome.blade.php [hunk 8]
```diff
@@ -1,45 +1,72 @@
...
+        }
+        .btn:hover {
+            background: #f51869;
+            background-image: -webkit-linear-gradient(top, #f51869, #f00000);
+            background-image: -moz-linear-gradient(top, #f51869, #f00000);
+            background-image: -ms-linear-gradient(top, #f51869, #f00000);
+            background-image: -o-linear-gradient(top, #f51869, #f00000);
+            background-image: linear-gradient(to bottom, #f51869, #f00000);
+            text-decoration: none;
+        }
```
Comment (n1c01a5) [inline resources/views/welcome.blade.php]: Add `<link href=style.css" rel="stylesheet" type="text/css">` with the style.

File: app/Http/Controllers/CustomerController.php (hunk 2 above)
Comment (brunobuddy) [inline app/Http/Controllers/CustomerController.php]: done

File: app/Http/Controllers/CustomerController.php (hunk 1 above)
Comment (brunobuddy) [inline app/Http/Controllers/CustomerController.php]: done

File: app/Http/routes.php (hunk 3 above)
Comment (brunobuddy) [inline app/Http/routes.php]: done

File: app/Services/ContactService.php (hunk 4 above)
Comment (brunobuddy) [inline app/Services/ContactService.php]: done

File: app/Http/Controllers/CustomerController.php (hunk 2 above)
Comment (n1c01a5) [inline app/Http/Controllers/CustomerController.php]: my previous comment is wrong. I think better is `abs($transactionAmount)`

File: app/Services/ContactService.php (hunk 5 above)
Comment (brunobuddy) [inline app/Services/ContactService.php]: exactement, done

File: config/settings.php (hunk 6 above)
Comment (brunobuddy) [inline config/settings.php]: done

File: database/migrations/2016_11_26_162123_create_transactions_table.php (hunk 7 above)
Comment (brunobuddy) [inline database/migrations/2016_11_26_162123_create_transactions_table.php]: done

File: resources/views/welcome.blade.php (hunk 8 above)
Comment (brunobuddy) [inline resources/views/welcome.blade.php]: ok

**Pull request merged**